### PR TITLE
fix(query): fix negative for u64

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

 fix negative for u64

- Closes #issue

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/12603)

### CODE REVIEW

**Review by lichuang — APPROVED** [review]

File: src/query/functions/src/scalars/arithmetic.rs [hunk 1]
```diff
@@ -498,6 +498,37 @@ fn register_unary_minus(registry: &mut FunctionRegistry) {
...
+                        },
+                        vectorize_with_builder_1_arg::<NumberType<u64>, NumberType<i64>>(
+                            |a, output, ctx| {
+                                let res = (a as i128).wrapping_neg();
+
+                                if res < std::i64::MIN as i128 || res > std::i64::MAX as i128 {
+                                    ctx.set_error(output.len(), "number overflowed");
+                                    output.push(0);
+                                } else {
+                                    output.push(res as i64);
```
Comment (andylokandy) [inline src/query/functions/src/scalars/arithmetic.rs:526]:
```suggestion
                                const MAX: u64 = -i64::MIN as u64;

                                if a > MAX {
                                    ctx.set_error(output.len(), "number overflowed");
                                    output.push(0);
                                } else {
                                    output.push(-(a as i64));
```

File: src/query/functions/src/scalars/arithmetic.rs [hunk 2]
```diff
@@ -517,13 +517,11 @@ fn register_unary_minus(registry: &mut FunctionRegistry) {
                         },
                         vectorize_with_builder_1_arg::<NumberType<u64>, NumberType<i64>>(
                             |a, output, ctx| {
-                                let res = (a as i128).wrapping_neg();
-
-                                if res < std::i64::MIN as i128 || res > std::i64::MAX as i128 {
+                                if a > -i64::MIN as u64 {
```
Comment (andylokandy) [inline src/query/functions/src/scalars/arithmetic.rs:520]: same as the domain calculation?

File: src/query/functions/src/scalars/arithmetic.rs (hunk 1 above)
Comment (andylokandy) [reply src/query/functions/src/scalars/arithmetic.rs]: oh, -i64::MIN will overflow...

**Comment by sundy-li** [comment]
```
 if a > -i64::MIN as u64 {
    |                              ^^^^^^^^^ attempt to negate `i64::MIN`, which would overflow
```

File: src/query/functions/src/scalars/arithmetic.rs [hunk 3]
```diff
@@ -498,6 +498,35 @@ fn register_unary_minus(registry: &mut FunctionRegistry) {
...
+                            }
+
+                            FunctionDomain::Domain(SimpleDomain::<i64> {
+                                min: min as i64,
+                                max: max as i64,
+                            })
+                        },
+                        vectorize_with_builder_1_arg::<NumberType<u64>, NumberType<i64>>(
+                            |a, output, ctx| {
+                                if a < std::i64::MIN as i128 || a > std::i64::MAX as i128 {
```
Comment (andylokandy) [inline src/query/functions/src/scalars/arithmetic.rs:520]:
```suggestion
                                if a < -(i64::MIN as i128) as u64 {
```
Eventually, llvm will fold it to a u64 constant.

**Pull request merged**